### PR TITLE
Feature/ctb 2/fetch data about the project from server

### PR DESCRIPTION
**Pull request by YauhenIlyin**

### CODE REVIEW

File: core/src/main/java/by/ilyin/core/controller/CargoTrackingCoreController.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import by.ilyin.core.service.AppInfoService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api")
+public class CargoTrackingCoreController {
```
Comment (Egor-Duvalov) [inline core/src/main/java/by/ilyin/core/controller/CargoTrackingCoreController.java]: Better AppInfoCoreController

File: core/src/main/java/by/ilyin/core/controller/CargoTrackingCoreController.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    private final AppInfoService appInfoService;
+
+    @GetMapping("/about")
+    public AppInfoDTO getAppInfo() {
+        System.out.println(appInfoService);
```
Comment (Egor-Duvalov) [inline core/src/main/java/by/ilyin/core/controller/CargoTrackingCoreController.java]: Remove all println comands before pushing

File: core/src/main/java/by/ilyin/core/service/AppInfoService.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package by.ilyin.core.service;
+
+import by.ilyin.core.entity.dto.AppInfoDTO;
+
+public interface AppInfoService {
```
Comment (Egor-Duvalov) [inline core/src/main/java/by/ilyin/core/service/AppInfoService.java:5]: Don't create interface, if you are not going create several implementations

File: web/src/main/java/by/ilyin/web/controller/CargoTrackingWebController.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import by.ilyin.web.service.AppInfoService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/api")
+public class CargoTrackingWebController {
```
Comment (Egor-Duvalov) [inline web/src/main/java/by/ilyin/web/controller/CargoTrackingWebController.java]: Better AppInfoWebController

File: web/src/main/resources/application.yaml [hunk 5]
```diff
@@ -1,2 +1,6 @@
 server:
-  port: 8080
\ No newline at end of file
+  port: 8080
+feign:
+  client:
+    core:
+      url: localhost:8090
```
Comment (Egor-Duvalov) [inline web/src/main/resources/application.yaml]: Create environment variables

File: web/src/main/java/by/ilyin/web/service/impl/AppInfoServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+@RequiredArgsConstructor
+@Service
+public class AppInfoServiceImpl implements AppInfoService {
+
+    private final AppInfoFeignClient appInfoFeignClient;
+
+    @Override
+    public AppInfoDTO getAppInfo() {
+        AppInfoDTO appInfoDTO;
+        appInfoDTO = appInfoFeignClient.getAboutInfo();
```
Comment (DzmitryKazakevich) [inline web/src/main/java/by/ilyin/web/service/impl/AppInfoServiceImpl.java]: @YauhenIlyin Method can be done in 1 line

File: web/src/main/java/by/ilyin/web/controller/CargoTrackingWebController.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+    private final AppInfoService appInfoService;
+
+    @GetMapping("/about")
+    public AppInfoDTO getAppInfo() {
+        AppInfoDTO appInfoDTO;
+        appInfoDTO = appInfoService.getAppInfo();
```
Comment (DzmitryKazakevich) [inline web/src/main/java/by/ilyin/web/controller/CargoTrackingWebController.java]: @YauhenIlyin Method can be done in 1 line

File: web/src/main/java/by/ilyin/web/service/feign/AppInfoFeignClient.java [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package by.ilyin.web.service.feign;
+
+import by.ilyin.web.entity.dto.AppInfoDTO;
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.web.bind.annotation.GetMapping;
+
+@FeignClient(name = "core", url = "${feign.client.core.url}")
```
Comment (DzmitryKazakevich) [inline web/src/main/java/by/ilyin/web/service/feign/AppInfoFeignClient.java]: @YauhenIlyin Core is not correct bean name

File: core/src/main/java/by/ilyin/core/controller/CargoTrackingCoreController.java (hunk 1 above)
Comment (DzmitryKazakevich) [reply core/src/main/java/by/ilyin/core/controller/CargoTrackingCoreController.java]: @Egor-Duvalov Please do not suggest some better name - just point that name should be better.
@YauhenIlyin FYI

File: core/src/main/java/by/ilyin/core/controller/CargoTrackingCoreController.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+        AppInfoDTO appInfoDTO;
+        appInfoDTO = appInfoService.getAppInfo();
```
Comment (DzmitryKazakevich) [inline core/src/main/java/by/ilyin/core/controller/CargoTrackingCoreController.java]: @YauhenIlyin Rewrite to 1 line

File: core/src/main/java/by/ilyin/core/service/impl/AppInfoServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+public class AppInfoServiceImpl implements AppInfoService {
+    @Value(value = "${app.info.applicationName}")
+    private String applicationName;
+    @Value(value = "${app.info.version}")
+    private String version;
+
+    @Override
+    public AppInfoDTO getAppInfo() {
+        AppInfoDTO appInfoDTO;
+        appInfoDTO = new AppInfoDTO(applicationName, version);
```
Comment (DzmitryKazakevich) [inline core/src/main/java/by/ilyin/core/service/impl/AppInfoServiceImpl.java]: @YauhenIlyin 1 line

**Review by Egor-Duvalov — APPROVED** [review]

**Pull request merged**